module/app/ante/sig_verify.go

```diff
@@ -0,0 +1,51 @@
+package ante
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+
+	ethermintante "github.com/evmos/ethermint/app/ante"
+)
+
+// Gravity supports both ordinary SDK signing (legacy amino and sign direct), but also has EIP712 support.
+// The Ethermint ante handler supports EVM messages and also maintains multiple lists of ante decorators,
+// which is undesireable for Gravity. Instead this decorator will delegate the signing to the correct
+// function depending on the type of transaction being processed.
+//
+// More accurately, if the input transaction has EXACTLY ONE ExtensionOptionsWeb3Tx option on it, we
```

Comment:
does the exactly one requirement here prevent multiple message tx? Not that I'm opposed to that restriction.

Reply:
That shouldn't impact it, this ExtensionOptionsWeb3Tx exists on the Tx, not the Msgs